src/client/Microsoft.Identity.Client/IConfidentialClientApplication.cs

```diff
@@ -86,6 +87,21 @@ public partial interface IConfidentialClientApplication : IClientApplicationBase
         /// URL of the authorization endpoint with the specified parameters.</returns>
         GetAuthorizationRequestUrlParameterBuilder GetAuthorizationRequestUrl(IEnumerable<string> scopes);
 
+        /// <summary>
+        /// Acquires a token without user interaction using username and password authentication.
+        /// This method does not look in the token cache, but stores the result in it. Before calling this method, use other methods 
+        /// such as <see cref="IClientApplicationBase.AcquireTokenSilent(IEnumerable{string}, IAccount)"/> to check the token cache.
+        /// </summary>
+        /// <param name="scopes">Scopes requested to access a protected API.</param>
+        /// <param name="username">Identifier of the user application requests token on behalf.
+        /// Generally in UserPrincipalName (UPN) format, e.g. <c>john.doe@contoso.com</c></param>
+        /// <param name="password">User password as a string.</param>
+        /// <returns>A builder enabling you to add optional parameters before executing the token request.</returns>
+        /// <remarks>
+        /// Available only for .NET Framework and .NET Core applications. See <see href="https://aka.ms/msal-net-up">our documentation</see> for details.
+        /// </remarks>
+        public AcquireTokenByUsernameAndPasswordConfidentialParameterBuilder AcquireTokenByUsernamePassword(IEnumerable<string> scopes, string username, string password);
```

Comment:
This is breaking change. You'll have to add it in a similar way you added `AcquireTokenByRefreshToken`